src/Faker/Container/ContainerBuilder.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Container;
+
+use Faker\English\FileExtension;
+use Faker\Extension\File;
+use Psr\Container\ContainerInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+class ContainerBuilder
+{
+    /**
+     * @var array
+     */
+    private $container = [];
+
+    /**
+     * @param string|callable|object $value
+     * @param string|null $name
+     */
+    public function add($value, string $name = null): self
+    {
+        if (!is_string($value) && !is_callable($value) && !is_object($value)) {
+            throw new \LogicException(sprintf('First argument to "%s::add()" must be a string, callable or object.', __CLASS__));
+        }
+
+        if ($name === null) {
+            if (is_string($value)) {
+                $name = $value;
+            } elseif (is_object($value)) {
+                $name = get_class($value);
+            } else {
+                throw new \LogicException(sprintf('Second argument to "%s::add()" is required not passing a string or object as first argument', __CLASS__));
```

Comment:
I know that Symfony likes to use single-line throws, but they are difficult to read. 

Can we spread the arguments to `sprintf()` across multiple lines>

Reply:
I like the fact that an exception message does not take more than one line. It makes the rest of the code easy to read. 

I'll update the code